main.tf

```diff
@@ -0,0 +1,314 @@
+    "google_pubsub_topic.topic",
+    "google_bigquery_dataset.dataset",
+  ]
+}
+
+# Billing-level
+# resource "google_logging_billing_account_sink" "sink" {
+#   count           = "${local.is_billing_level ? 1 : 0}"
+#   name            = "${var.name}"
+#   billing_account = "${var.billing_id}"
+#   filter          = "${var.filter}"
+#   destination     = "${local.destination_uri}"
+#
+#   depends_on = [
+#     "google_storage_bucket.bucket",
+#     "google_pubsub_topic.topic",
+#     "google_bigquery_dataset.dataset",
+#   ]
+# }
+
+#------------------------------#
+# Service account IAM bindings #
+#------------------------------#
+
+# Org-level
+resource "google_project_iam_member" "sink_member_org" {
+  count   = "${local.is_org_level ? 1 : 0}"
+  project = "${local.destination_project}"
+  role    = "${local.role}"
+  member  = "${google_logging_organization_sink.sink.writer_identity}"
+
+  depends_on = [
+    "google_logging_organization_sink.sink",
+  ]
+}
+
+# Folder-level
+resource "google_project_iam_member" "sink_member_folder" {
+  count   = "${local.is_folder_level ? 1 : 0}"
+  project = "${local.destination_project}"
+  role    = "${local.role}"
+  member  = "${google_logging_folder_sink.sink.writer_identity}"
+
+  depends_on = [
+    "google_logging_folder_sink.sink",
+  ]
+}
+
+# Project-level
+resource "google_project_iam_member" "sink_member_project" {
+  count   = "${local.is_project_level ? 1 : 0}"
+  project = "${local.destination_project}"
+  role    = "${local.role}"
+  member  = "${google_logging_project_sink.sink.writer_identity}"
+
+  depends_on = [
+    "google_logging_project_sink.sink",
+  ]
+}
+
+# Billing-level
+# resource "google_project_iam_member" "sink_member_billing_account" {
+#   count   = "${local.is_billing_level ? 1 : 0}"
+#   project = "${local.destination_project}"
+#   role    = "${local.role}"
+#   member  = "${google_logging_billing_account_sink.sink.writer_identity}"
+#
+#   depends_on = [
+#     "google_logging_billing_account_sink.sink",
+#   ]
+# }
+
+#-----------------------------------------------#
+# Pub/Sub topic subscription (for integrations) #
+#-----------------------------------------------#
+resource "google_service_account" "pubsub_subscriber" {
+  count        = "${lookup(local.destination, "create_subscriber", false) ? 1 : 0}"
+  account_id   = "${local.destination_name}-subscriber"
+  display_name = "${local.destination_name} Topic Subscriber"
+  project      = "${local.destination_project}"
+}
+
+resource "google_project_iam_binding" "pubsub_subscriber_binding" {
+  count   = "${lookup(local.destination, "create_subscriber", false) ? 1 : 0}"
+  role    = "roles/pubsub.subscriber"
+  project = "${local.destination_project}"
+
+  members = [
+    "serviceAccount:${google_service_account.pubsub_subscriber.email}",
+  ]
+
+  depends_on = [
+    "google_service_account.pubsub_subscriber",
+  ]
+}
+
+resource "google_project_iam_binding" "pubsub_viewer_binding" {
```

Comment:
Can't this binding happen on the pubsub topic itself instead of the whole project?

Reply:
Yep ! I think I missed the existence of https://www.terraform.io/docs/providers/google/r/pubsub_topic_iam.html